gpflow/base.py

```diff
@@ -59,17 +61,29 @@ def __init__(self,
             self._unconstrained = tf.Variable(value, dtype=dtype, name=name, trainable=trainable)
 
         self.prior = prior
+        self.prior_on_x = prior_on_x
         self._transform = transform
 
-    def log_prior(self):
+    def log_prior(self, evaluate_on_x: bool = False):
+        """ Prior probability density.
+        This can be evaluated either on the constrained or unconstrained variable. """
```

Comment:
Can you add a bit more explanation to the docstring what the difference is ?

When would you use evaluate_on_x=True ?

Reply:
Yep have updated docstring with a worked example. 

I've now set the default to True to resolve https://github.com/GPflow/GPflow/issues/1176.  It's also useful to try both in order to expose the sensitivity of MAP to the choice of parameterisation.